### PR TITLE
Fix mementos for data management | Add pushMenu to NSpecDAO

### PR DESCRIPTION
**Pull request by sarthak-marwaha**

### CODE REVIEW

File: src/foam/nanos/boot/NSpec.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ foam.CLASS({
   ],
 
   imports: [
-    'ctrl'
+    'ctrl', 'pushMenu'
```
Comment (Nauna) [inline src/foam/nanos/boot/NSpec.js]: should be on a new line

File: src/foam/nanos/boot/NSpec.js [hunk 2]
```diff
@@ -278,7 +278,7 @@ foam.CLASS({
       code: function() {
         var service = this.__context__[this.name];
         if ( foam.dao.DAO.isInstance(service) ) {
-          this.ctrl.memento.value = 'admin.data' + foam.nanos.controller.Memento.SEPARATOR + this.name;
+          this.pushMenu(foam.nanos.menu.Menu.create({ id: 'admin.data' + foam.nanos.controller.Memento.SEPARATOR + this.name }));
```
Comment (Nauna) [inline src/foam/nanos/boot/NSpec.js]: hmm the applicationController.pushMenu - checks if the menu through the menuDAO to confirm you have permission to that menu - so creating a menu here isn't going to work.
However applicationController.pushMenu will consider the memento.value being updated ... so don't think this change will be beneficial.

Comment (sarthak-marwaha) [reply src/foam/nanos/boot/NSpec.js]: Does the aplicationController.pushMenu listen to memento changes?

Comment (sarthak-marwaha) [reply src/foam/nanos/boot/NSpec.js]: Updated

**Review by Nauna — APPROVED** [review]

**Pull request merged**